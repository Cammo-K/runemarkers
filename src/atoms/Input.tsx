import { useContext, useState } from "react";
import { SearchContext } from "../App";
import styles from "./Input.module.css";

export default function Input() {
  const [searching, setSearching] = useState(false);
  const [searchVal, setSearchVal] = useContext(SearchContext);

  return (
    <input
      type="text"
      placeholder={searching ? "" : "type here to search"}
      className={styles.input}
      onFocus={() => setSearching(true)}
      onBlur={() => {
        if (searchVal === "") {
          setSearching(false);
        }
      }}
      style={{
        width: searching ? 250 : 175,
        minWidth: 50,
        maxWidth: 500,
      }}
      onChange={(e) => {
        if (setSearchVal) {
          setSearchVal(e.target.value);
        }
      }}
    />
  );
}
