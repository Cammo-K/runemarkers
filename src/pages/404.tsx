import NavBar from '@/components/molecules/NavBar';
import styles from '@/styles/404.module.css';
import { NextSeo } from 'next-seo';
import Head from 'next/head';
import Image from 'next/image';
import Link from 'next/link';

export default function FourOhFour() {
  return (
    <>
      <Head>
        <link href="runescape_uf.ttf" as="font" type="font/woff2" />
        <NextSeo title="404" description="Page not found" />
      </Head>
      <NavBar />
      <div className={styles.container}>
        <h1 className={styles.header}>404</h1>
        <div className={styles.imageContainer}>
          <Image
            className={styles.image}
            src={`https://thumbs.gfycat.com/SecondaryUncommonKestrel.webp`}
            alt="404 gif"
            width={270}
            height={490}
            layout="intrinsic"
            objectFit="contain"
            priority={true}
          />
        </div>
        <Link href="/">
          <span className={styles.home}>Go Home</span>
        </Link>
      </div>
    </>
  );
}
