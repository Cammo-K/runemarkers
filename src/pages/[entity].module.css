.container {
  margin-top: 80px;
  display: flex;
  justify-content: center;
  align-items: center;
  height: 100%;
  padding: 20px;
}

@media (max-width: 768px) {
  .container {
    max-height: none;
  }
}

@media (max-width: 400px) {
  .container {
    margin-top: 80px;
  }
}

.link {
  color: var(--color-primary);
  font-size: 1rem;
  margin: 10px 10px 10px 10px;
  transition: all 0.2s ease-out;
}

.link:hover {
  color: var(--color-primary-light);
}

.linkContainer {
  display: flex;
  justify-content: center;
  align-items: center;
}

.tagContainer > p > span {
  color: #aaa;
  transition: all 0.2s ease-out;
}

.tagContainer > p {
  margin: 0 5px 0 0;
}

.tagContainer {
  display: flex;
  align-items: center;
  justify-content: center;
  color: #aaa;
}
