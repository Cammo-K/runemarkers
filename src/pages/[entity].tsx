import { TileEntity } from '@/types';
import { NextSeo } from 'next-seo';
import NavBar from '@/components/molecules/NavBar';
import styles from '@/pages/[entity].module.css';
import ListContainer from '@/components/atoms/ListContainer';
import TileEntityCard from '@/components/molecules/TileEntityCard';
import YoutubeEmbed from '@/components/atoms/YouTubeEmbed';
import CodeBlock from '@/components/atoms/CodeBlock';
import ContributionFooter from '@/components/atoms/ContributionFooter';
import { getTileData } from '@/api/tiles';
import { defaultImages } from '@/api/seoOptions';
import ListContainerSection from '@/components/molecules/ListContainerSection';
import TilesSource from '@/components/molecules/TilesSource';

export async function getStaticPaths() {
  return {
    paths: getTileData().map((e) => {
      return {
        params: { entity: e.safeURI },
      };
    }),
    fallback: false,
  };
}

export async function getStaticProps({
  params,
}: {
  params: { entity: string };
}) {
  return {
    props: getTileData().find(
      (e) => e.safeURI.toLowerCase() === params.entity.toLowerCase(),
    ),
  };
}

export default function Entity(entity: TileEntity) {
  return (
    <>
      <NextSeo
        title={`${entity.name}${
          entity.subcategory ? ` (${entity.subcategory}) ` : ` `
        }Tile Markers`}
        description={`${entity.name}${
          entity.altName ? ` / ${entity.altName}` : ``
        } tile markers for RuneLite. Find and import tile markers for different Oldschool RuneScape activities.`}
        openGraph={{
          images: [
            {
              url: entity.thumbnail,
              alt: `${entity.name} tile markers`,
            },
            ...defaultImages,
          ],
        }}
      />
      <NavBar />
      <div className={styles.container}>
        <ListContainer>
          <TileEntityCard entity={entity} hideInfoButton />
          <div className={styles.linkContainer}>
            <a
              href={`https://runelite.net/tile/show/#${Buffer.from(
                JSON.stringify(entity.tiles),
              )
                .toString(`base64`)
                .replaceAll(`=`, ``)}`}
              className={styles.link}
              target="_blank"
              rel="noopener noreferrer"
            >
              View Map on RuneLite
            </a>
            <a
              href={entity.wiki}
              className={styles.link}
              target="_blank"
              rel="noopener noreferrer"
            >
              Wiki Page
            </a>
          </div>
          {entity.recommendedGuideVideoId ? (
            <ListContainerSection title="Recommended Guide">
              <YoutubeEmbed
                videoId={entity.recommendedGuideVideoId}
                title={`${entity.name} guide`}
              />
            </ListContainerSection>
          ) : null}
          <ListContainerSection title="Tile Data">
            <CodeBlock tiles={entity.tiles} />
          </ListContainerSection>

          {entity.source ? (
            <ListContainerSection title="Source">
              <TilesSource source={entity.source} />
            </ListContainerSection>
          ) : null}
        </ListContainer>
      </div>
      <ContributionFooter />
    </>
  );
}
