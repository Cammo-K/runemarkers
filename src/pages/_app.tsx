import { AppProps } from 'next/app';
import Head from 'next/head';
import '@/styles/global.css';
import { DefaultSeo } from 'next-seo';
import nextSeoOptions from '@/api/seoOptions';
import Script from 'next/script';

export default function MyApp({ Component, pageProps }: AppProps) {
  return (
    <>
      {process.env.NEXT_PUBLIC_GA_TRACKING_ID ? (
        <>
          <Script
            id="google-analytics"
            async
            src={`https://www.googletagmanager.com/gtag/js?id=${process.env.NEXT_PUBLIC_GA_TRACKING_ID}`}
            strategy="lazyOnload"
          />
          <Script id="google-analytics-script" strategy="lazyOnload">
            {`
          window.dataLayer = window.dataLayer || [];
          function gtag(){dataLayer.push(arguments);}
          gtag('js', new Date());
          gtag('config', '${process.env.NEXT_PUBLIC_GA_TRACKING_ID}');
          `}
          </Script>
        </>
      ) : null}
      <DefaultSeo {...nextSeoOptions} />
      <Head>
        <meta name="viewport" content="initial-scale=1.0, width=device-width" />
        <meta name="author" content="RuneMarkers" />
        <meta charSet="utf-8" />
        <meta name="viewport" content="initial-scale=1.0, width=device-width" />
        <meta name="theme-color" content="#262624" />
      </Head>
      <Component {...pageProps} />
    </>
  );
}
