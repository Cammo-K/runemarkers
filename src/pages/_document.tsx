import { Html, Head, Main, NextScript } from 'next/document';

export default function Document() {
  return (
    <Html>
      <Head>
        <meta name="author" content="RuneMarkers" />
        <meta property="og:site_name" content="RuneMarkers" />
        <meta property="og:type" content="website" />
        <meta property="og:url" content="https://runemarkers.net" />
        <meta property="og:image" content="/logo1024.png" />
        <meta charSet="utf-8" />
        <meta name="theme-color" content="#262624" />
        <link
          rel="apple-touch-icon"
          sizes="180x180"
          href="/apple-touch-icon.png"
        />
        <link
          rel="icon"
          type="image/png"
          sizes="32x32"
          href="/favicon-32x32.png"
        />
        <link
          rel="icon"
          type="image/png"
          sizes="16x16"
          href="/favicon-16x16.png"
        />
        <link rel="manifest" href="/site.webmanifest.json" />
      </Head>
      <body>
        <Main />
        <NextScript />
      </body>
    </Html>
  );
}
