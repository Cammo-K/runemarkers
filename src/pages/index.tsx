import Head from 'next/head';
import Fuse from 'fuse.js';
import { TileEntity } from '@/types';
import tileJson from '../tiles.json';
import { createContext, useEffect, useState } from 'react';
import TileEntityList from '@/components/organisms/TileEntityList';
import NavBar from '@/components/molecules/NavBar';
import styles from '@/styles/Home.module.css';
// import TileEntityList from "./organisms/TileEntityList";

const fuseOptions = {
  // debugging
  // includeScore: true,
  // the keys of the objects to search
  keys: [
    { name: `name`, weight: 0.7 },
    { name: `tags`, weight: 0.3 },
  ],
  threshold: 0.4,
};

// Assert the structure of the tileJson
const tileData: TileEntity[] = tileJson;

const fuse = new Fuse(tileData, fuseOptions);

export const SearchContext = createContext<
  [string | undefined, React.Dispatch<React.SetStateAction<string>> | undefined]
>([undefined, undefined]);

export default function Home() {
  const [searchVal, setSearchVal] = useState(``);
  const [searchRes, setSearchRes] = useState<TileEntity[]>(tileData);

  useEffect(() => {
    if (searchVal.length > 0) {
      setSearchRes(fuse.search(searchVal).map((res) => res.item));
      return;
    }
    setSearchRes(tileData);
  }, [searchVal]);
  return (
    <div>
      <Head>
        <title>RuneMarkers - A Collection of RuneLite Ground Markers</title>
        <meta
          name="description"
          content="Find and import RuneLite ground markers for oldschool runescape. Easily search for ground markers by boss, activity or location."
        />
        <meta name="author" content="RuneMarkers" />
        <meta property="og:site_name" content="RuneMarkers" />
        <meta
          property="og:title"
          content="RuneMarkers - A Collection of RuneLite Ground Markers"
        />
        <meta
          property="og:description"
          content="Find and import RuneLite ground markers for oldschool runescape."
        />
        <meta property="og:type" content="website" />
        <meta property="og:url" content="https://runemarkers.net" />
        <meta property="og:image" content="/logo1024.png" />
        <meta charSet="utf-8" />
        <meta name="viewport" content="initial-scale=1.0, width=device-width" />
        <meta name="theme-color" content="#262624" />
        <link
          rel="apple-touch-icon"
          sizes="180x180"
          href="/apple-touch-icon.png"
        />
        <link
          rel="icon"
          type="image/png"
          sizes="32x32"
          href="/favicon-32x32.png"
        />
        <link
          rel="icon"
          type="image/png"
          sizes="16x16"
          href="/favicon-16x16.png"
        />
        <link rel="manifest" href="/site.webmanifest" />
      </Head>

      <main className={styles.main}>
        <SearchContext.Provider value={[searchVal, setSearchVal]}>
          <NavBar />
          <TileEntityList list={searchRes} />
        </SearchContext.Provider>
      </main>
    </div>
  );
}
