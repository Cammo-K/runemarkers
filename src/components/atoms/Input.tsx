import { SearchContext } from '@/pages';
import styles from '@/components/atoms/Input.module.css';
import { useContext, useState } from 'react';

export default function Input() {
  const [searching, setSearching] = useState(false);
  const [searchVal, setSearchVal] = useContext(SearchContext);

  return (
    <input
      type="text"
      placeholder={searching ? `` : `type here to search`}
      className={styles.input}
      onFocus={() => setSearching(true)}
      onBlur={() => {
        if (searchVal === ``) {
          setSearching(false);
        }
      }}
      style={{
        width: searching ? 250 : 175,
        minWidth: 50,
        maxWidth: 500,
      }}
      onChange={(e) => {
        if (setSearchVal) {
          setSearchVal(e.target.value);
        }
      }}
    />
  );
}
