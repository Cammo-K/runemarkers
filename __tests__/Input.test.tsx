import { SearchContext } from '@/pages';
import '@testing-library/jest-dom';
import { act, fireEvent, render } from '@testing-library/react';
import { Dispatch, SetStateAction } from 'react';
import Input from '@/components/atoms/Input';

const testInputWithContext = (
  searchVal: string | undefined,
  setSearchVal: Dispatch<SetStateAction<string>>,
) => (
  <SearchContext.Provider value={[searchVal, setSearchVal]}>
    <div>
      <Input />
      <button></button>
    </div>
  </SearchContext.Provider>
);
describe(`Input`, () => {
  it(`should render correctly`, () => {
    const { container } = render(<Input />);
    expect(container).toMatchSnapshot();
  });

  it(`should expand on focus`, () => {
    const mockSetSearchVal = jest.fn();

    jest.mock(`react`, () => ({
      useState: (initial: string) => [initial, mockSetSearchVal],
    }));

    const { container } = render(testInputWithContext(``, mockSetSearchVal));

    const input = container.querySelector(`input`);
    if (!input) {
      throw new Error(`No input found`);
    }
    act(() => {
      fireEvent.focus(input);
    });
    expect(input.placeholder).toBe(``);
  });

  it(`should shrink on blur`, () => {
    const mockSetSearchVal = jest.fn();

    jest.mock(`react`, () => ({
      useState: (initial: string) => [initial, mockSetSearchVal],
    }));

    const { container } = render(testInputWithContext(``, mockSetSearchVal));

    const input = container.querySelector(`input`);
    if (!input) {
      throw new Error(`No input found`);
    }
    act(() => {
      fireEvent.focus(input);
    });
    expect(input.placeholder).toBe(``);
    act(() => {
      input.value = ``;
      fireEvent.blur(input);
    });
    expect(input.placeholder).toBe(`type here to search`);
  });

  it(`should update the search value on change`, () => {
    const mockSetSearchVal = jest.fn();

    jest.mock(`react`, () => ({
      useState: (initial: string) => [initial, mockSetSearchVal],
    }));

    const { container } = render(testInputWithContext(``, mockSetSearchVal));

    const input = container.querySelector(`input`);
    if (!input) {
      throw new Error(`No input found`);
    }
    act(() => {
      fireEvent.focus(input);
    });
    expect(input.placeholder).toBe(``);
    act(() => {
      fireEvent.change(input, { target: { value: `test` } });
    });
    expect(mockSetSearchVal).toHaveBeenCalledTimes(1);
    expect(mockSetSearchVal).toHaveBeenCalledWith(`test`);
  });
});
