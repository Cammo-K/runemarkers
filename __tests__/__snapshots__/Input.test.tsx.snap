// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should render correctly 1`] = `
<div>
  <input
    class="input"
    placeholder="type here to search"
    style="width: 175px; min-width: 50px; max-width: 500px;"
    type="text"
  />
</div>
`;
